src/sage/modular/modform_hecketriangle/hecke_triangle_groups.py

```diff
@@ -1028,18 +1032,12 @@ def is_cycle_of_length(seq, n):
             for el in self._conj_block[t_ZZ]:
                 group_el = prod([self.V(el[k][0])**el[k][1] for k in range(len(el))])
 
-                #if el != group_el.conjugacy_type():
-                #    raise AssertionError("This shouldn't happen!")
-
                 D = group_el.discriminant()
-                if coerce_AA(D) < 0:
-                    raise AssertionError("This shouldn't happen!")
-                if coerce_AA(D) == 0:
+                if coerce_AA(D) <= 0:
                     raise AssertionError("This shouldn't happen!")
```

Comment:
Wouldn't this be better as
```suggestion
                assert coerce_A4(0) > 0
```